lib/release/release.go

```diff
@@ -0,0 +1,204 @@
+	TLSConfig *tls.Config
+	// ReleaseServerAddr is the address of the release server
+	ReleaseServerAddr string
+}
+
+// CheckAndSetDefaults checks and sets default config values
+func (c *ClientConfig) CheckAndSetDefaults() error {
+	if c.TLSConfig == nil {
+		return trace.BadParameter("missing TLS configuration")
+	}
+
+	if c.ReleaseServerAddr == "" {
+		return trace.BadParameter("missing release server address")
+	}
+
+	return nil
+}
+
+// Client is a client to make HTTPS requests to the
+// release server using the Teleport Enterprise license
+// as authentication
+type Client struct {
+	// client is the client used to make calls to the release API
+	client *roundtrip.Client
+}
+
+// NewClient returns a new release client with a client
+// to make https requests to the release server
+func NewClient(cfg ClientConfig) (*Client, error) {
+	err := cfg.CheckAndSetDefaults()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: cfg.TLSConfig,
+		},
+	}
+
+	client, err := roundtrip.NewClient(fmt.Sprintf("https://%s", cfg.ReleaseServerAddr), "", roundtrip.HTTPClient(httpClient))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &Client{
+		client: client,
+	}, nil
+}
+
+// ListRelease calls the release server and returns a list of releases
+func (c *Client) ListReleases(ctx context.Context) ([]*types.Release, error) {
+	if c.client == nil {
+		return nil, trace.BadParameter("client not initialized")
+	}
+
+	resp, err := c.client.Get(ctx, c.client.Endpoint(types.EnterpriseReleaseEndpoint), nil)
+	if err != nil {
+		log.WithError(err).Error("failed to retrieve releases from release server")
+		return nil, trace.Wrap(err)
+	}
+
+	if resp.Code() == http.StatusUnauthorized {
+		return nil, trace.AccessDenied("access denied by the release server")
+	}
+
+	if resp.Code() != http.StatusOK {
+		return nil, trace.Errorf("release server responded with status %d", resp.Code())
+	}
+
+	var releases []Release
+	err = json.Unmarshal(resp.Bytes(), &releases)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var responseReleases []*types.Release
+	for _, r := range releases {
+		var releaseAssets []*types.Asset
+		for _, a := range r.Assets {
+			releaseAssets = append(releaseAssets, &types.Asset{
+				Arch:        a.Arch,
+				Description: a.Description,
+				Name:        a.Name,
+				OS:          a.OS,
+				SHA256:      a.SHA256,
+				AssetSize:   a.Size,
+				DisplaySize: byteCount(a.Size),
+				ReleaseIDs:  a.ReleaseIDs,
+				PublicURL:   a.PublicURL,
+			})
+		}
+
+		responseReleases = append(responseReleases, &types.Release{
+			NotesMD:   r.NotesMD,
+			Product:   r.Product,
+			ReleaseID: r.ReleaseID,
+			Status:    r.Status,
+			Version:   r.Version,
+			Assets:    releaseAssets,
+		})
+	}
+
+	return responseReleases, err
+}
+
+// byteCount converts a size in bytes to a human-readable string.
+func byteCount(b int64) string {
```

Comment:
nit: I would consider moving this function to one of the  teleport common utils package